ENS system tests

nikoulai opened the pull request:
## Description

#5039 

## Type of change

System tests for ENS package

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

nikoulai committed:
Add initial setup

nikoulai committed:
Add system utils symlink

nikoulai committed:
Registry: change variable name in setsubnodeowner and pass context object to contract

nikoulai committed:
Add contract artifacts

nikoulai committed:
Fix contracts setup

nikoulai committed:
Finish tests

nikoulai committed:
Remove unused registar and reverse registar

nikoulai committed:
Delete setUpENS

nikoulai committed:
Tests refactoring

nikoulai committed:
Merge branch '4.x' into nikos/5039/ens-4.x-system-tests

nikoulai committed:
Add accounts.json

nikoulai committed:
Fix unit test

nikoulai committed:
Add context in registry

nikoulai committed:
Use different builds for ens internal contracts

nikoulai committed:
Revert files deletion

nikoulai committed:
Add ws provider

nikoulai committed:
Fix conditional provider

jdevcs commented on packages/web3-eth-ens/src/abi/ens/ENSRegistry.ts:
```diff
@@ -0,0 +1,395 @@
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+export const ENSRegistryAbi = [
```
We already have registry https://github.com/ChainSafe/web3.js/blob/4.x/packages/web3-eth-ens/src/abi/registry.ts , why do we need this new file?

nikoulai replied:
For running the tests, I needed the bytecode of the contracts so I compiled the ones mentioned in the headers. Initially, I wanted to be sure that I had the abi of the exactly same version.

However, when I tried to restore to the initial abis (the ones you mention) I was getting an error (TypeError: abis is not iterable). I'll keep investigating the error.

avkos replied:
If you need bytecode only for the tests - please add them to `test/fixtures` folder

jdevcs commented on packages/web3-eth-ens/src/abi/ens/PublicResolver.ts:
```diff
@@ -0,0 +1,894 @@
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+export const PublicResolverAbi = [
```
Same for resolver, there is already resolver ABI https://github.com/ChainSafe/web3.js/blob/4.x/packages/web3-eth-ens/src/abi/resolver.ts

nikoulai committed:
Remove try catch

nikoulai committed:
Increase gas to block gas limit

nikoulai committed:
Merge branch '4.x' into nikos/5039/ens-4.x-system-tests

nikoulai committed:
Fix test script

nikoulai committed:
Restore try catch

nikoulai committed:
Add ipc provider

nikoulai committed:
Create dedicated bytecode files for tests

avkos commented on packages/web3-eth-ens/package.json:
```diff
@@ -22,11 +22,12 @@
 		"test:watch": "npm test -- --watch",
 		"test:unit": "jest --config=./test/unit/jest.config.js",
-		"test:integration": "jest --config=./test/integration/jest.config.js --passWithNoTests --runInBand",
+		"test:integration": "jest --config=./test/integration/jest.config.js --runInBand --forceExit",
```
Is it possible to make tests without `--forceExit`?
if you need to close a connection we have a beautiful function `closeOpenConnection` from `system_tests_utils`

nikoulai replied:
I missed closing some providers, thanks! Fixed!

Muhammad-Altabba commented on packages/web3-eth-ens/src/abi/ens/DummyNameWrapper.ts:
```diff
@@ -0,0 +1,40 @@
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+export const DummyNameWrapperApi = [
```
If you could move this, please, to `packages/web3-eth-ens/test/fixtures` similar to how it is done in other packages. Like for example: `packages/web3-eth-contract/test/fixtures/erc721.ts`.

nikoulai replied:
In this package, the abis are used also in the class. The whole `ens` class is a wrapper of these smart contracts.
 (cc: @avkos )

Muhammad-Altabba replied:
Thanks for the clarification.
And so I hope to have a better name for that. Because the word "Dummy" is not very instructive. Or what do you think?

nikoulai replied:
You are right about that. The reason I gave that name, is to point to the right [contract](https://github.com/ensdomains/ens-contracts/tree/master/contracts/resolvers/mocks) (which I used) and avoid any misunderstandings. What do you think, should I rename it?

Muhammad-Altabba replied:
Yes, please, as I think the name Dummy miss-leaded me a little :blush:

nikoulai committed:
Remove forceExit

nikoulai committed:
Merge branch '4.x' into nikos/5039/ens-4.x-system-tests

nikoulai committed:
detectOpenHandles

nikoulai committed:
close connection for all providers

nikoulai committed:
close connection for all providers for IPC

Muhammad-Altabba reviewed (approved):
Besides the other review comments, the MR looks great to me.

nikoulai committed:
Rename DummyNameWrapper

jdevcs commented on packages/web3-eth-ens/test/fixtures/ens/bytecode/PublicResolverBytecode.ts:
```diff
@@ -0,0 +1,18 @@
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
```
we should add link to source code of contract here, from which this byte code was generated. ( for making it easier to debug in future )

jdevcs commented on packages/web3-eth-ens/test/fixtures/ens/bytecode/NameWrapperBytecode.ts:
```diff
@@ -0,0 +1,18 @@
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+export const NameWrapperBytecode =
```
we should add link to source code of contract here, from which this byte code was generated. ( for making it easier to debug in future )

jdevcs commented on packages/web3-eth-ens/test/fixtures/ens/bytecode/ENSRegistryBytecode.ts:
```diff
@@ -0,0 +1,18 @@
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+export const ENSRegistryBytecode =
```
we should add link of source code of contract here in comments, from which this byte code was generated. ( for making it easier to debug in future )

jdevcs reviewed (commented):
LGTM, just one item to be changed:
There are already registry.ts, and resolver.ts having ABIs, we should have one version of each contract ABI instead of two versions of each. Also in ABI + byte code files, add contract source link in comments.

nikoulai committed:
Various fixes

- Add link to source code for bytecode and abis
- Update json bytecode
- Fix NameWrapper typo
